src/dymaptic.GeoBlazor.Core/Components/Views/MapView.razor.cs

```diff
@@ -2264,19 +2264,31 @@ protected List<string> GetActiveEventHandlers()
 
         foreach (PropertyInfo callbackInfo in callbacks)
         {
-            dynamic? callback = callbackInfo.GetValue(this);
+            object? callback = callbackInfo.GetValue(this);
```

Comment:
@nwestfall this is great! I have a few suggestions on how we can make it more streamlined.

- On line 2260, we could actually pull the `EventCallback`s and `Func`s separately as propertyInfos, instead of together. This would eliminate the need to check if it's a Func later.
- There is an interface, `IEventCallback` that all the `EventCallback`s and `EventCallback<T>`s inherit from. I think we could just cast to that instead of each return type. This would make it more extensible, as with the current version, if we add new EventCallbacks with different types, we have to remember to update this method.

I think I'd like to see a refactor using one or both of these ideas before we merge. Let me know if you would like me to implement that myself, or you are welcome to update this branch and PR.

Thanks again for finding this and working on it!

Reply:
@TimPurdum  I'll look at your first point.  For the second point, I actually tried this but it's an internal interface.

https://github.com/dotnet/aspnetcore/blob/1fbe264736d62fab1c37f0986df2e90fd4ab33b3/src/Components/Components/src/IEventCallback.cs#L7

Reply:
Ah, I missed that. Bummer!

Reply:
If we can't find a way around enumerating the callback `T`s, I'm ok with keeping it.

Reply:
The only way I could think of doing it (and maybe this is longer term), is if all the GeoBlazor Events (`ClickEvent`, `BlurEvent`), implemented an interface.  Would have to test, but maybe then you could do `EventCallback<IGeoBlazorEvent>` to catch all of them.

Reply:
Doesn't look like you can :(, so I don't have any other ideas at this time.